frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -233,5 +234,21 @@ def rescue_from_spree_gateway_error(exception)
     def check_authorization
       authorize!(:edit, current_order, cookies.signed[:guest_token])
     end
+
+    def insufficient_stock_error
+      packages = @order.shipments.map(&:to_package)
+      if packages.empty?
+        flash[:error] = I18n.t('spree.insufficient_stock_for_order')
+        redirect_to cart_path
+      else
+        availability_validator = Spree::Stock::AvailabilityValidator.new
+        unless @order.line_items.all? { |item| availability_validator.validate(item) }
+          unavailable_items = @order.line_items.map(&:name).to_sentence
```

Comment:
`@order.line_items.all? { |item| availability_validator.validate(item) }` means that at least one line item is not valid, right? So, I'm not sure that `unavailable_items = @order.line_items.map(&:name).to_sentence` is right because it lists all line items as unavailable. Perhaps

```ruby
unavailable_items = @order.line_items.select { |line_item| !availability_validator.validate(line_item) }.map(&:name).to_sentence
```

would do (haven't tested). WDYT?

Reply:
You're right, something went wrong with my cut & paste refactoring. It should be fine now, thank you for spotting the bug 🐛